### PR TITLE
Restore PHP 5.3/5.4 compat

### PR DESCRIPTION
**Pull request by dvdoug**
Behat 3.6 has broken support for PHP <5.5 even though `composer.json` is still claiming to support PHP >= 5.3.3.

This PR hopefully fixes that and adds PHP 5.3, 5.4 and 5.5 to CI as it currently doesn't seem to test these older versions 🤔 

Fully behind you guys dropping support for older PHP in the future, just please update `composer.json` first 😉

### CODE REVIEW

**Comment by dvdoug** [comment]
Scrutinizer failure is a false-positive, there's nothing I can do about that. I've reported it, and they say there's already a ticket their side for it.

This should be good to go

**Comment by upyx** [comment]
May be is it the time to drop support of old PHP versions?

**Comment by dvdoug** [comment]
> 
> 
> May be is it the time to drop support of old PHP versions?

Need to fix it first though, otherwise running `composer update` breaks projects (I discovered this through a broken build)

**Comment by upyx** [comment]
> Scrutinizer failure is a false-positive, there's nothing I can do about that.

It failures because of the first '\\' symbol. It should be skipped in fully-qualified class names:
```php
interface_exists('Symfony\\Component\\Translation\\TranslatorInterface')
```

**Comment by dvdoug** [comment]
The leading slash is equivalent to not having one, see rule #1 on https://www.php.net/manual/en/language.namespaces.rules.php. Happy to remove though.

The failure on line 10 is because the interface exists on the version of Symfony that isn't installed on Scrutinzer and the static analysis isn't picking up on the fact that it's guarded by `interface_exists` on line 9.

File: .travis.yml [hunk 1]
```diff
@@ -14,6 +14,15 @@ branches:
 
 matrix:
   include:
+    - php: 5.3
+      dist: precise
+      env: DEPENDENCIES='low'
+    - php: 5.4
+      dist: trusty
+      env: DEPENDENCIES='low'
```
Comment (stof) [inline .travis.yml]: I suggest removing these `DEPENDENCIES='low'`. We should rather test on higher (compatible) versions than on lowest ones, to match what our users will get (we need one job using `DEPENDENCIES='low'` in our matrix, but if you look at it, you will see that it also has other normal jobs running on that PHP version)

Comment (dvdoug) [reply .travis.yml]: Done

**Comment by ciaranmcnulty** [comment]
I updated from master and the 5.3 build doesn't seem to be working

**Comment by dvdoug** [comment]
@ciaranmcnulty Traits are 5.4 and https://github.com/Behat/Behat/pull/1270/files seems to have removed that restriction from the test suite

**Comment by dvdoug** [comment]
I'm not sure there's an elegant way of excluding that test file from the 5.3 build using the revised tag system, I'm thinking it might be easier to simply `rm` it inside Travis on the 5.3 build? How would maintainers feel about that?

Another idea would be adding `Given my PHP version is >= x.y` as step definition to the test suite but that seems like a lot of extra code for the purpose of skipping 1 test.

Or...could temporarily revert #1270, release a final 5.3 supporting version and then put #1270 back on master alongside a bumped minimum PHP version?

**Comment by Sam-Burns** [comment]
@dvdoug This approach seems to be passing on Travis, if it is of any interest https://github.com/Sam-Burns/Behat/compare/master...Sam-Burns:restore_php54_compat

**Comment by Sam-Burns** [comment]
Would be good to get your fix included, and do a patch release with working PHP 5.3 support. After speaking to @ciaranmcnulty, it sounds like it may well be the last release to support EOL PHP versions 🍾

**Comment by dvdoug** [comment]
> 
> 
> Would be good to get your fix included, and do a patch release with working PHP 5.3 support. After speaking to @ciaranmcnulty, it sounds like it may well be the last release to support EOL PHP versions 🍾

Awesome @Sam-Burns, I've cherry-picked that one into my branch and rebased on master.

**Comment by Sam-Burns** [comment]
Scrutinizer is complaining about a reference to a nonexistent interface, although frankly that is the entire point of the relevant file. I'd suggest ignoring it, if it's possible to merge the PR anyway.

**Comment by dvdoug** [comment]
> 
> 
> Scrutinizer is complaining about a reference to a nonexistent interface, although frankly that is the entire point of the relevant file. I'd suggest ignoring it, if it's possible to merge the PR anyway.

Yep, it's a false positive dating back to before this PR. Any chance this could get merged/released @ciaranmcnulty?

**Comment by pamil** [comment]
Fixes #1280.

**Comment by pamil** [comment]
I've rebased it with master, but it's failing on one of the builds connected to the container interop. @dvdoug, could you please take a look at it?

**Comment by pamil** [comment]
Thanks, Doug! :tada:

**Pull request merged**